src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -149,6 +201,8 @@ public void removeTaskId(String taskId) {
         if (entry.getValue() != null && entry.getValue().equals(taskId)) {
           journalNodes.put(entry.getKey(), null);
           setJournalNodes(journalNodes);
+          Date date = DateUtils.addSeconds(new Date(), conf.getDeadNodeTimeout());
+          deadJournalNodeTimeStamp = new Timestamp(date.getTime());
```

Comment:
I wanted to keep the code as clean as possible. Basically it will update the time stamp if another node fails, so it will also go off of the timestamp of the last failure (it does this based on node type JN, NN, ZKFC). This would cover the cases adequately and there shouldn't be a bug. Can you think of any scenario where this would be a problem?